### PR TITLE
Inkscape tool revamp and gamnit support

### PR DESCRIPTION
**Pull request by xymus**
* Clean up old style inits, add missing doc and refactor main code into classes and methods.
* Adds the gamnit target option, the generated code is much more simple, but the generation a bit more complex.
* Allow to specify the full path to the Nit source file to generate.
* Make rounding to the next power of 2 optional, it will no more be required for mnit programs on Android with #1284.

### CODE REVIEW

**Comment by privat** [comment]
+1

File: contrib/inkscape_tools/README.md [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
+It selects objects to extract from the SVG file when their id begins with `0`.
 
-# Features
+It will produce two files:
 
-* Creates a sinlge PNG file per SVG source file
-* Creates subimages for objects with an id beginning by 0.
-* If the id ends with 0 to 9, will instead create an array of subimages.
+* A Nit source file that declares a single class with an attribute for each selected object.
+  The attribute usually holds a singles texture, except if the the id ends with a digit, then it will be an array.
```
Comment (Morriar) [inline contrib/inkscape_tools/README.md]: single*

File: contrib/inkscape_tools/src/svg_to_png_and_nit.nit [hunk 2]
```diff
@@ -14,51 +14,140 @@
...
 
-# This script extracts pngs from a single svg for all objects with ids
-# beginning by 0. Requires Inkscape.
+# Extract images of objects from an SVG file using Inkscape
 module svg_to_png_and_nit
 
 import opts
 import template
 
+# Image info extracted from the SVG file
```
Comment (Morriar) [inline contrib/inkscape_tools/src/svg_to_png_and_nit.nit]: infos* or data*

File: contrib/inkscape_tools/src/svg_to_png_and_nit.nit [hunk 3]
```diff
@@ -14,51 +14,140 @@
...
+	# Source maximum X
+	var max_x: Int
+
+	# Source minimum Y
+	var min_y: Int
+
+	# Source maximum Y
+	var max_y: Int
+
+	# Get the coordinates for `image` as `"x, y, w, h"`
```
Comment (Morriar) [inline contrib/inkscape_tools/src/svg_to_png_and_nit.nit]: Is there a particular use case for that string? Why not returning an array?

File: contrib/inkscape_tools/src/svg_to_png_and_nit.nit [hunk 4]
```diff
@@ -14,51 +14,140 @@
...
 
 	redef fun rendering
 	do
+		# Known array of images
+		var arrays_of_images = new Array[String]
+
+		# Attributes of the class
+		var attributes = new Array[String]
+
+		# Statements for the `load_all_` method
```
Comment (Morriar) [inline contrib/inkscape_tools/src/svg_to_png_and_nit.nit]: The only `load_all_` method I can see if in fact named `load_all` without the last underscore. Are talking about other methods?

**Comment by Morriar** [comment]
+1

File: contrib/inkscape_tools/src/svg_to_png_and_nit.nit (hunk 3 above)
Comment (xymus) [inline contrib/inkscape_tools/src/svg_to_png_and_nit.nit]: It is used only as a string for the generated code.

**Pull request merged**